In [ ]:
import numpy as np
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
df = pd.read_csv('//content//drive//MyDrive//Colab Notebooks//News headlines Jan 17 - Apr 21.csv')
num_rows_org = df.shape[0]
print(num_rows_org)

200497


In [ ]:
pattern = '[^\x00-\x7F]+'

filtered_df = df[~df['Title'].str.contains(pattern)]

# Save the filtered DataFrame back to a CSV file
filtered_df.to_csv('/content/drive/MyDrive/Colab Notebooks/eng_headlines.csv', index=False)

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/eng_headlines.csv')

In [ ]:
# Converting Date column to datetime in the format DD/MM/YYYY - allows for better analysis of date-related data
df['Date'] = pd.to_datetime(df['Date'], format='%d-%m-%Y')

# Group headlines by date and keep only the top 60 headlines from each day
df_filter = df.groupby('Date').head(60)

# Save the filtered DataFrame back to a CSV file
df_filter.to_csv('/content/drive/MyDrive/Colab Notebooks/eng_headlines_60.csv', index=False)

# Opening our NEW FILTERED DATASET WHICH CONTAINS ENG HEADLINES (max 60 for each date)
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/eng_headlines_60.csv')

#no. of rows in our new dataset
num_rows = df.shape[0]
print('No. of Rows = ', num_rows)

#count of headlines for each date after keeping the max 50 constraint
headlines_per_date = df['Date'].value_counts()

#checking no. of dates with less than 10 headlines
less10 = (headlines_per_date < 10).sum()
print('No. of dates with less than 10 headlines = ', less10)

No. of Rows =  89663
No. of dates with less than 10 headlines =  3


In [ ]:
senti_counts = df['sentiment'].value_counts()
print(senti_counts)

NEGATIVE    47835
POSITIVE    41828
Name: sentiment, dtype: int64


After our basic filteration process, our New Dataset is eng_top60.csv
It has 89663 rows. 47835 of these rows contain NEGATIVE sentiment label whereas 41828 contain POSITIVE sentiment label.
